cognite/client/_api/postgres_gateway/users.py

```diff
@@ -0,0 +1,274 @@
+        limit: int | None = None,
+    ) -> Iterator[User] | Iterator[UserList]:
+        """Iterate over users
+
+        Fetches user as they are iterated over, so you keep a limited number of users in memory.
+
+        Args:
+            chunk_size (int | None): Number of users to return in each chunk. Defaults to yielding one user a time.
+            limit (int | None): No description.
+
+        Returns:
+            Iterator[User] | Iterator[UserList]: yields User one by one if chunk_size is not specified, else UserList objects.
+        """
+        self._warning.warn()
+
+        return self._list_generator(
+            list_cls=UserList,
+            resource_cls=User,
+            method="GET",
+            chunk_size=chunk_size,
+            limit=limit,
+            headers={"cdf-version": "beta"},
+        )
+
+    def __iter__(self) -> Iterator[User]:
+        """Iterate over users
+
+        Fetches users as they are iterated over, so you keep a
+        limited number of users  in memory.
+
+        Returns:
+            Iterator[User]: yields user one by one.
+        """
+        return self()
+
+    @overload
+    def create(self, user: UserWrite) -> User: ...
+
+    @overload
+    def create(self, user: Sequence[UserWrite]) -> UserList: ...
+
+    def create(self, user: UserWrite | Sequence[UserWrite]) -> User | UserList:
+        """`Create Users <https://api-docs.cognite.com/20230101-beta/tag/Postgres-Gateway-Users/operation/create_users>`_
+
+        Create postgres users.
+
+        Args:
+            user (UserWrite | Sequence[UserWrite]): None
+
+        Returns:
+            User | UserList: A user
+
+        Examples:
+
+            Create user:
+
+                >>> import os
+                >>> from cognite.client import CogniteClient
+                >>> from cognite.client.data_classes.postgres_gateway import UserWrite, SessionCredentials
+                >>> from cognite.client.data_classes import ClientCredentials
+                >>> client = CogniteClient()
+                >>> session = client.iam.sessions.create(
+                ...     ClientCredentials(os.environ.get("IDP_CLIENT_ID"), os.environ.get("IDP_CLIENT_SECRET")),
+                ...     session_type="CLIENT_CREDENTIALS"
+                ... )
+                >>> user = UserWrite(credentials=SessionCredentials(nonce=session.nonce))
+                >>> res = client.postgres_gateway.users.create(user)
+
+        """
+        self._warning.warn()
+        return self._create_multiple(
+            list_cls=UserList,
+            resource_cls=User,
+            items=user,
+            input_resource_cls=UserWrite,
+            headers={"cdf-version": "beta"},
+        )
+
+    @overload
+    def update(self, items: UserUpdate) -> User: ...
+
+    @overload
+    def update(self, items: UserWrite) -> User: ...
+
+    def update(self, items: UserUpdate | UserWrite | Sequence[UserUpdate | UserWrite]) -> User | UserList:
```

Comment:
missing overloads for Sequence. And having separate overloads for `UserUpdate` and `UserWrite` serves no purpose.